ANFOG/DM/anfog_dm_netcdf_compliance.sh

```diff
@@ -0,0 +1,146 @@
+        sed -e 's/^.*HEAD$//g' | sed -e 's/\t//g'  | sed -e '/^$/d' | tr '\n' ' ' )"
+
+    local var
+    for var in $varlist_3d; do
+        nc_set_att -a coordinates,$var,o,c,'TIME LATITUDE LONGITUDE DEPTH' $nc_file
+    done
+
+    declare -a varlist_2d=("[U-V]CUR" "[U-V]CUR_GPS" "VBSC" "PROFILE" "PHASE" "HEAD")
+
+    for var in $varlist_2d; do
+        nc_set_att -a coordinates,$var,o,c,'TIME LATITUDE LONGITUDE' $nc_file
+    done
+}
+
+# fix qc variables that have unknown standard name
+# # $1 - netcdf file
+nc_remove_att_from_qc_variables() {
+    local nc_file=$1; shift
+    local qc="quality_control"
+    local varlist="IRRAD[[:digit:]]{3}_${qc} [U-V]CUR_GPS_${qc} VBSC_${qc} PROFILE_${qc} PHASE_${qc} HEAD_${qc} CDOM_${qc}"
+
+    local var
+    for var in $varlist; do
+        var=`nc_list_variables $nc_file | grep "$var" | head -1`
+        if [ x"$var" != x ]; then
+            nc_set_att -a standard_name,$var,d,, $nc_file
+        fi
+    done
+}
+
+#######################################
+# IMOS function to fix ANFOG DM files #
+#######################################
+
+# make changes to pass IMOS compliance checks
+# $1 - netcdf file
+fix_imos_conventions() {
+    local nc_file=$1; shift
+
+    # global attributes
+    nc_set_att -a data_centre,global,o,c,"${DATA_CENTRE}" $nc_file
+    nc_set_att -a acknowledgement,global,o,c,"${ACKNOWLEDGEMENT}" $nc_file
+    nc_set_att -a distribution_statement,global,o,c,"${DISTRIBUTION_ST}" $nc_file
+    nc_set_att -a geospatial_lat_units,global,o,c,"degrees_north" $nc_file
+    nc_set_att -a geospatial_lon_units,global,o,c,"degrees_east" $nc_file
+    nc_set_att -a geospatial_vertical_units,global,o,c,"meter" $nc_file
+    nc_set_att -a naming_authority,global,o,c,"IMOS" $nc_file
+    nc_set_att -a project,global,o,c,"${PROJECT}" $nc_file
+
+    # variables
+    nc_set_att -a units,TIME,o,c,"days since 1950-01-01 00:00:00 UTC" $nc_file
+    nc_set_att -a ancillary_variables,PHASE,o,c,"PHASE_quality_control PROFILE" $nc_file
+    nc_set_att -a ancillary_variables,PROFILE,o,c,"PROFILE_quality_control PHASE" $nc_file
+    nc_set_att -a ancillary_variables,UCUR_GPS,o,c,"UCUR_GPS_quality_control" $nc_file
+
+    fix_imos_qc_convention $nc_file
+    fix_var_long_name $nc_file
+    fix_depth_min_max $nc_file
+}
+
+fix_imos_qc_convention() {
+    local nc_file=$1; shift
+    nc_set_att -a quality_control_conventions,_quality_control,o,c,"${CONVENTIONS}" $nc_file
+}
+
+fix_var_long_name() {
+    nc_file=$1; shift
+    local varlist="SENSOR[[:digit:]] DEPLOYMENT PLATFORM"
+
+    local var
+    for var in $varlist; do
+        var=`nc_list_variables $nc_file | grep "$var" | head -1`
+        if [ x"$var" != x ]; then
+            nc_set_att -a long_name,$var,o,c,"${var,,} informations" $nc_file
+        fi
+    done
+}
+
+fix_depth_min_max() {
+    nc_file=$1; shift
+    # val.nc WTF?!
+    local depth_min=`ncap2 -O -C -v -s "val=DEPTH.min();print(val)" ${nc_file} ~/val.nc | cut -f 3- -d ' ' ;`
```

Comment:
this is a compulsory argument -output- of the ncap2 function.  I should add a line to delete straight after extracting the min max value. we can rename it 'tmp' if you prefer